lib/ash_authentication/strategies/magic_link.ex

```diff
@@ -191,12 +191,15 @@ defmodule AshAuthentication.Strategy.MagicLink do
   """
   def request_token_for_identity(strategy, identity, opts \\ [], context \\ %{})
       when is_struct(strategy, __MODULE__) do
+        context_map = Map.from_struct(context)
+        source_context = Map.get(context_map, :source_context, %{})
+        context_claims = Map.get(source_context, :claims, %{})
```

Comment:
Hm...I think we'd have to stew on this one. Likely if we do this the name here would need to be much more specific though, like `:magic_link_token_claims` for example.

@jimsynz thoughts?